source/common/router/config_impl.cc

```diff
@@ -528,7 +539,8 @@ RouteEntryImplBase::RouteEntryImplBase(const VirtualHostImpl& vhost,
   }
 
   for (const auto& mirror_policy_config : route.route().request_mirror_policies()) {
-    shadow_policies_.push_back(std::make_shared<ShadowPolicyImpl>(mirror_policy_config));
+    shadow_policies_.push_back(
```

Comment:
nit: (not because of your PR)
pre-allocate vector length (`reserve()`) to size of `route.route().request_mirror_policies()`

Reply:
Good point. Done